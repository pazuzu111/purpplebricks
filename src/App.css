body {
  background: #ED213A;  /* fallback for old browsers */
background: -webkit-linear-gradient(to right, #93291E, #ED213A);  /* Chrome 10-25, Safari 5.1-6 */
background: linear-gradient(to right, #93291E, #ED213A); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
font-family: 'Quicksand';
}

.App {
  text-align: center;
}

.App-logo {
  height: 50px;
}

.App-header {
  background: #ED213A;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #93291E, #ED213A);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #93291E, #ED213A); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  height: 43px;
  padding: 20px;
  padding-right: 41px;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

#back {
  height: 35vw;
  cursor: pointer;
}

#search {
  border-radius: 6px;
  border: none;
  height: 15px;
  width: 210px;
  font-size: 13px;
  padding: 15px 7px;
  outline: none;
  -webkit-transition: .3s ease-in-out;
  transition: .3s;
}

#search:focus {
  background: red;
  color:white;
  transition: .3s ease-in-out;
}

form {
  margin-left: 33px;
}

.filter-wrapper {
  display: flex;
  align-items: center;
  margin-left: auto;
}

.App-header input {
  border-radius: 20px;
  border: none;
  height: 50px;
  width: 400px;
  font-size: 21px;
  padding: 5px 15px;
  outline: none;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.subtype-filter {
  padding: 20px;
  color: grey;
  cursor: pointer;
  transition: all .2s ease-in-out;
}

.subtype-filter img {
  height: 30px;
}

.subtype-filter p {
  margin: 0;
  color: white;
}

.subtype-filter:hover {
  background: gold;
  transition: all .2s ease-in-out;
}
.loader {
  width: 80%;
  margin-top: 40px;
}

.card-container {
  width: 95%;
  margin: 0 auto;
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
}

.card {
  height: 250px;
  margin: 30px 10px;
  position: relative;
  transition: all .3s ease-in-out;
}

.card:hover {
  transform: scale(1.1);
  transition: all .3s ease-in-out;
}

.card img {
  height: 100%;
  transition: .3s all ease-out
}

.card h2 {
  position: relative;
  bottom: 120px;
  transform: rotate(-25deg);
  color: green;
  background: white;
  border-radius: 12px;
}

.selected-wrapper {
  display: flex;
  padding: 15px;
  justify-content: center;
  flex-wrap: wrap;
}

.img-wrapper {
  height: 400px;
  margin: 10px;
}

.info-wrapper {
  margin: 10px;
  color: #fff;
  padding: 20px 75px;
}

.img-wrapper img {
  height: 100%;
}


@media (max-width: 520px) {
  .App-header input {
    width: 80%;
  }

  .card-container {
    flex-direction: column;
  }

  .card {
    height: 350px;
  }

  .card button {
    width: 54%;
  }

  #back {
    height: 35vw;
  }
}

@media (max-width: 700px) {
  form, form input {
    width: 100% !important;
  }

  #back {
    height: 50vw;
  }
}

@media (max-width: 861px) {
  .filter-wrapper {
    height: 100%;
    width: 100%;
    padding: 0 15px;
    justify-content: space-evenly;
  }

  .subtype-filter img {
    height: 30px !important;
  }

  .App-header {
    flex-direction: column;
    height: 100%;
  }

  #back {
    height: 35vw;
  }
}
