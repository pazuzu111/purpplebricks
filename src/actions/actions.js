import {fetchData} from '../services/fetchService'

export const GET_CARDS = 'GET_CARDS'
export const getCards = (name) => dispatch => {
    dispatch({
        type: 'GET_CARDS',
        name
    })
}

export const RECEIVE_CARDS = 'RECEIVE_CARDS'
export const receiveCards = (name, data) => dispatch => {
    dispatch({
        type: 'RECEIVE_CARDS',
        name,
        cards: data
    })
}

//action creator
export const getTheCards = (name) => dispatch => {
    dispatch(getCards(name))
    fetchData(name).then(cards => dispatch(receiveCards(name, cards)))
}

