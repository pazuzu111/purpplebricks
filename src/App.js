import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getTheCards, filterTheCards } from './actions/actions'
import Navbar from './components/Navbar'
import Cards from './components/Cards'

import './App.css';

class App extends Component {
  constructor(props) {
    super(props)
  }

  componentDidMount() {
    this.searchPokemon('')
  }

  searchPokemon = name => { this.props.getTheCards(name) }

  filterPokemon = name => { this.props.filterTheCards(name) }

  render() {
    return (
      <div className="App">
          <Navbar handleSearch={this.searchPokemon} filterPokemon={this.filterPokemon} />
          <Cards loading={this.props.loading} cards={this.props.cards} />           
      </div>
    );
  }
}

const mapStateToProps = state => {
  return { cards: state.cards, loading: state.loading };
};

const mapDispatchToProps = dispatch => {
    return {
      getTheCards: name => dispatch(getTheCards(name)),
      filterTheCards: type => dispatch(filterTheCards(type))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
