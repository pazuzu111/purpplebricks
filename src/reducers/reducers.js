import { GET_CARDS, RECEIVE_CARDS, FILTER_CARDS } from '../actions/actions'

const initialState = {
    cards: []
}
export default (state = initialState, action) => {
    switch (action.type) {
        case GET_CARDS:
        return { ...state, loading: true }

        case RECEIVE_CARDS:
        console.log("DATA HERE!!", action.cards)
        let data = action.cards
        return { ...state, cards: data, loading: false }

        case FILTER_CARDS:
        return { ...state, loading: true }
        
        default: return state
    }
}