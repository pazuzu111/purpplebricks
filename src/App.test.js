import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import reducers from './reducers/reducers'
import { GET_CARDS, RECEIVE_CARDS, FILTER_CARDS } from './actions/actions';
import * as actions from './actions/actions'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

const cards = [{data: {title: 'one'}}]

describe('fetch reducer', () => {
  it('has a default state', () => {
    expect(reducers(undefined, {}))
    .toMatchSnapshot()
  })
  
  it('handles GET_CARDS request', () => {
    expect(reducers(undefined, {
      type:   GET_CARDS,
      payload: {
       name: ''
      }
    })).toEqual({
      loading: true,
      cards: []
    })
  })

  it('handles RECEIVE_CARDS request', () => {
    expect(reducers([], {
      type:   RECEIVE_CARDS,
      payload: {
        name: '',
        cards: cards
      }
    })).toEqual({
      loading: false
    })
  })

  it('handles FILTER_CARDS request', () => {
    expect(reducers(undefined, {
      type:   FILTER_CARDS,
      payload: {
        pokemonType: 'mega'
      }
    })).toEqual({
      loading: true,
      cards: []
    })
  } )
})

describe('testing getCards action', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore();
  
  beforeEach(() => {
    store.clearActions();
  });

  it('should dispatch the getCards action and payload', () => {
    const expectedAction = [{ 'name': '', 'type': 'GET_CARDS' },]

    store.dispatch(actions.getCards(''))
    expect(store.getActions()).toEqual(expectedAction)
  })

  it('should dispatch the receiveCards action and payload', () => {
    const expectedAction = [{ 'name': '', 'type': 'RECEIVE_CARDS', cards: cards },]

    store.dispatch(actions.receiveCards('', cards))
    expect(store.getActions()).toEqual(expectedAction)
  })

  it('should dispatch the filterCards action and payload', () => {
    const expectedAction = [{ 'pokemonType': 'mega', 'type': 'FILTER_CARDS'}]

    store.dispatch(actions.filterCards('mega'))
    expect(store.getActions()).toEqual(expectedAction)
  })


})