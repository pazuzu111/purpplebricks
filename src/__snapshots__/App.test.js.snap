// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch reducer has a default state 1`] = `
Object {
  "cards": Array [],
}
`;
